bayes_opt.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+from scipy.optimize import minimize
+from sklearn.gaussian_process import GaussianProcessRegressor
+from sklearn.gaussian_process.kernels import Matern
+
+
+class BayesianOptimizer(object):
+
+    def __init__(self, feature_meta,
```

Comment:
(by uniformly, I just mean, pulling randomly over-and-over from the bag without any explicit bias for those in higher density areas. I think especially because we're making the 100 dimensional space from word vectors but our documents don't have that many observations, making a KDE in that space will suffer from the dimensionality, so it will prolly be a useless KDE. I see no downside in sampling randomly from the data except that it will explore the nether regions less, but that's probably a fine bias...